include/rocksdb/options.h

```diff
@@ -1192,6 +1192,16 @@ struct DBOptions {
   //
   // Default: hostname
   std::string db_host_id = kHostnameForDbHostId;
+
+  // Use this if your DB want to enable checksum handoff for specific file
+  // types writes. Make sure that the File_system you use support the
+  // crc32c checksum verification
+  // Currently supported file tyes: kWALFile, kTableFile, kDescriptorFile.
+  // NOTE: currently RocksDB only generates crc32c based checksum for the
+  // handoff. If the storage layer has different checksum support, user
+  // should enble this vector is empty. Otherwise,it may cause unexpected
+  // write failures.
+  std::vector<FileType> checksum_handoff_file_types;
```

Comment:
vector is not a good abstraction for a set from a pretty small domain (11 values currently, so a set fits in uint16_t), and ties us to certain inefficiencies long term because this is public API. Inefficiency is not just membership query, but also copying DBOptions.

I'm envisioning a template<ENUM_TYPE,MAX_VALUE> class SmallEnumSet that is a wrapper around an unsigned scalar type of sufficient size to have one bit for each possible enum value (up to 64 for now, expandable in the future), given the largest enum value as a template parameter and assuming values are from 0 to that largest value. You could then have

    using FileTypeSet = SmallEnumSet<FileType, kBlobFile>;

Reply:
Yeah, it is more efficient not to use the vector. So is SmallEnumSet already in hour code base?

Reply:
No, it would need to be implemented, but for now it should only need Add and Contains.

Reply:
> No, it would need to be implemented, but for now it should only need Add and Contains.

Sure, will add it later to include/rocksdb/utilities